### PR TITLE
Add image replacement feature

### LINKED ISSUE #130: Feature: Replace File option in Media Library
Add option to 'Replace file' similar to Enable Media Replace but with Azure Blob container. 

https://wordpress.org/plugins/enable-media-replace/

### LINKED ISSUE #182: Add a 'Replace Media File' option
### Is your enhancement related to a problem? Please describe.

Allow users to replace/re-upload file. This is helpful when needing to update PDF file version or replace image in Media Library. 

### Designs

_No response_

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #184: WP 6.2.2 error
### Describe the bug

Hello, thank you for this awesome plugin, I've found some issues with image uploading with this: 

PHP: 8.2
WP: 6.2.2

I'm able to sync previous images with Azure, but can't add more images to Azure when I activate the Azure's plugin.

1172 line is about array_flip, but I'm pretty sure that the error is related with next line where blob returns an error.

I hope you can help me!

### Steps to Reproduce

Try to upload a photo to media library. After configuration.

### Screenshots, screen recording, code snippet

Stack trace:
#0 /home/ubuntu/environment/project/wp-content/plugins/windows-azure-storage/includes/class-windows-azure-rest-api-client.php(1172): array_flip()
#1 /home/ubuntu/environment/project/wp-content/plugins/windows-azure-storage/includes/class-windows-azure-rest-api-client.php(868): Windows_Azure_Rest_Api_Client->_sanitize_remote_paths()
#2 /home/ubuntu/environment/project/wp-content/plugins/windows-azure-storage/includes/class-windows-azure-helper.php(449): Windows_Azure_Rest_Api_Client->sanitize_blobs_names()
#3 /home/ubuntu/environment/project/wp-content/plugins/windows-azure-storage/windows-azure-storage.php(614): Windows_Azure_Helper::get_unique_blob_name()
#4 /home/ubuntu/environment/project/wp-includes/class-wp-hook.php(308): windows_azure_storage_wp_handle_upload_prefilter()
#5 /home/ubuntu/environment/project/wp-includes/plugin.php(205): WP_Hook->apply_filters()
#6 /home/ubuntu/environment/project/wp-admin/includes/file.php(810): apply_filters()
#7 /home/ubuntu/environment/project/wp-admin/includes/file.php(1082): _wp_handle_upload()
#8 /home/ubuntu/environment/project/wp-admin/includes/media.php(303): wp_handle_upload()
#9 /home/ubuntu/environment/project/wp-admin/includes/ajax-actions.php(2598): media_handle_upload()
#10 /home/ubuntu/environment/project/wp-admin/async-upload.php(33): wp_ajax_upload_attachment()
#11 {main}
  thrown in /home/ubuntu/environment/project/wp-content/plugins/windows-azure-storage/includes/class-windows-azure-rest-api-client.php on line 1172

### Environment information

PHP: 8.2
WP: 6.2.2

### WordPress information

6.2.2

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hugosolar**
### Description of the Change
Extending our previous feature to replace PDFs in #220 
Now I'm adding the feature to replace images.
The criteria to replace images is based on the source image, we're using it as a reference to handle metadata
The process is to loop over the registered sizes for the image and find the image with the nearest size from the replacement and replace the correspondent image size.
This approach seems to be the one used by known plugins like Enable Media replace

The workflow we're following is

- Selecting an image to be replaced
- Click "Replace this media"
- Select an image to replace
- The new image is going to be uploaded to Azure
- After that if "Replace media" is clicked:
  -  Replacing the main image
  - Loop through the source sizes in `_wp_attached_file` and copy those over to replace the location finding the closest size possible
  - Then we delete the newest attachment 

https://drive.google.com/file/d/1vPuk_cYUdWrlGBY3bUkjxWTpZeYr-BTG/view

This plugin also fixes #184 ensuring we're processing an array with array_flip otherwise returning a WP_Error object
closes #182 #130

### How to test the Change
- Clone this branch to your local environment
- Follow the workflow in the video: https://drive.google.com/file/d/1vPuk_cYUdWrlGBY3bUkjxWTpZeYr-BTG/view

Feature to replace images at the blob storage level

> Fixed - Bug fix
Fixed bug relate to the use of array_flip and not ensuring it was an actual arrat

### Credits
Props @hugosolar 

### Checklist:
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Comment by jeffpaul** [comment]
@rickalee looking to you / your team for review / approval here

**Review by rickalee — DISMISSED** [review]

**Comment by rickalee** [comment]
@hugosolar Approved. Looks great. Any considerations with Cache and configuration we need to document for other users of public plugin?

**Comment by hugosolar** [comment]
@rickalee I've addressed cache issue by setting the blob properties cache to 30 segs
Screenshot 2024-06-17 at 14 42 01

**Review by rickalee — DISMISSED** [review]

**Comment by hugosolar** [comment]
@jeffpaul the e2e error seems to come directly from `wp-env`
```
ERROR: for f505fb82f2963a5299b2972f47ad8834_mysql_1  'ContainerConfig'
Creating f505fb82f2963a5299b2972f47ad8834_tests-wordpress_1 ... done
Creating f505fb82f2963a5299b2972f47ad8834_tests-cli_1       ... 
Creating f505fb82f2963a5299b2972f47ad8834_tests-cli_1       ... done
[2378] Failed to execute script docker-compose

ERROR: for mysql  'ContainerConfig'
```

have you seen this before?
actually, we're using `@wordpress/env` `8.7.0` in the plugin and it's currently at `2.0.0` I can try to update that library and see how it goes

**Comment by dkotter** [comment]
@hugosolar Yes, updating `@wordpress/env` to the latest should fix that particular issue, as we've ran across this on a few other plugins. Seems GitHub updated the `ubuntu-latest` image and this causes issues with wp-env if it's not running the latest

**Comment by hugosolar** [comment]
@dkotter @jeffpaul I've updated `wp-env` and I can see the build is passing now
now it's failing on cypress tests since I'm pushing from a forked repository and secrets aren't available to accomplish those tests.
This has been fixed on my other PRs
let me know how can I help with this

File: includes/class-windows-azure-helper.php
Comment (dkotter) [inline includes/class-windows-azure-helper.php:516]: Since we're adding a new function param here, we should document that in the docblock

File: includes/class-windows-azure-replace-media.php
Comment (dkotter) [inline includes/class-windows-azure-replace-media.php:59]: Can we add a docblock on this property?

File: includes/class-windows-azure-replace-media.php
Comment (dkotter) [inline includes/class-windows-azure-replace-media.php:83]: Not sure if we do this consistently in this plugin but would be great to add proper hook documentation here: https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#4-hooks-actions-and-filters

File: includes/class-windows-azure-replace-media.php
Comment (dkotter) [inline includes/class-windows-azure-replace-media.php]: Can we remove this?

File: includes/class-windows-azure-replace-media.php
Comment (dkotter) [inline includes/class-windows-azure-replace-media.php:403]: Wondering if we need to be more defensive here? For both `$source_sizes` and `$target_sizes`, will these always be arrays and will the keys we want always exist? Or do we need to check for those before proceeding?

File: includes/class-windows-azure-replace-media.php
Comment (dkotter) [inline includes/class-windows-azure-replace-media.php]: I know this is not part of this PR but does this need to fire on the `nopriv` hook? Seems like replacing media is something that should only be triggered by a logged in user

**Review by dkotter — CHANGES_REQUESTED** [review]

File: includes/class-windows-azure-replace-media.php
Comment (hugosolar) [reply includes/class-windows-azure-replace-media.php:403]: @dkotter since this array is made above, it stores WordPress default `_wp_attachment_metadata` so it's always expected to exist if it's a valid attachment. Although I agree with the extra check because we never knows if we're about to handle an invalid attachment or a similar situation. 
I've added some extra checks to bail early

**Review by dkotter — DISMISSED** [review]

**Comment by jeffpaul** [comment]
Note I'm holding for @rickalee's review before merging here

**Review by rickalee — APPROVED** [review]

**Comment by rickalee** [comment]
@jeffpaul @hugosolar @dkotter Approved after a few updates by Hugo.

**Comment by jeffpaul** [comment]
@hugosolar mind checking on the e2e failures before this gets merged in?

**Comment by hugosolar** [comment]
@jeffpaul as far as I can see, it's failing because I'm pushing this PR from my fork
and it doesn't have environment keys for Blob storage cypress tests. Tried to add those environment variables to my fork but doesn't seem to be working.

**Pull request merged**
